mlos_bench/mlos_bench/environment/base_environment.py

```diff
@@ -58,7 +69,12 @@ def new(cls, env_name, class_name, config, global_config=None, tunables=None, se
         return instantiate_from_config(cls, class_name, env_name, config,
                                        global_config, tunables, service)
 
-    def __init__(self, name, config, global_config=None, tunables=None, service=None):
+    def __init__(self,
+                 name: str,
+                 config: dict,
+                 global_config: Optional[dict] = None,
+                 tunables: Optional[TunableGroups] = None,
+                 service: Optional[Service] = None):
         # pylint: disable=too-many-arguments
```

Comment:
Not a huge fan, mostly because from a developer standpoint we lose autocomplete and other easy quick checks on what are viable/expected arguments.
This is really just a pylint style warning.  PEP8 or whatever standard likes slightly more concise argument lists, but I don't think it's a hard requirement.  I don't consider it a safety thing like static type checking capabilities.

Reply:
Oh, sorry, I misinterpreted this earlier.  I thought you meant to replace all arguments with `*` and rely on processing `kwargs` imperatively.
Nm, I like this approach.  Forces `param=value` everywhere, which is much more readable.